### DISCUSSION TITLE
Battery voltage measurement

### DISCUSSION THREAD

**Discussion by icho40** [arendst/Tasmota#22050] [Q&A]
I would like to program a battery-operated ESP32:
(https://wiki.dfrobot.com/SKU_DFR1075_FireBeetle_2_Board_ESP32_C6)

"FireBeetle 2 ESP32-C6 integrates lithium battery charging management, 
allowing the lithium battery to be charged via Type-C, 5V DC, and solar panels."

I can compile these Tasmota source files:
https://github.com/arendst/Tasmota/tree/development/tasmota

This works well on my ESP32.

However, I would like the battery voltage to be displayed.
This is what I found: 
https://wiki.dfrobot.com/SKU_DFR1075_FireBeetle_2_Board_ESP32_C6_Basic_Tutorial#target_1

I added the code to tasmota.ino and the compilation ran without errors.
In the web interface I have set GPIO0 as 
“ADC VOLTAGE” in the web interface.
Unfortunately, it always shows me the same value, 
regardless of whether a battery is connected or not.

Also the values defined in the ino file “analogValue” and ‘analogVolts’ do not appear in the console.

Can someone help me?

**Comment by sfromis** [comment]
Not having the board, I'd not expect a zero when not connecting a battery, as it may be trying to charge a battery when you have USB connected. As a starting point, I'd disconnect USB before looking at the value.

You did not say what value you saw, better post details instead of vague words like "same".

There were pretty recent changes around how ADC is handled, better use latest development binary.

You also have the relatively new command `AdcGpio`, which you can use to customize details, after already having a value, with same linear fitting as the older `AdcParam` command.

**Reply by icho40** [reply]
> Not having the board, I'd not expect a zero when not connecting a battery, as it may be trying to charge a battery when you have USB connected. As a starting point, I'd disconnect USB before looking at the value.

USB was disconnected.

> You did not say what value you saw, better post details instead of vague words like "same".

I configured 2 GPIOs, GPIO0 and GPIO1 as "ADC Voltage".
On GPIO0 is the battery connected, on GPIO1 nothing.
ADC1 Voltage 2V
ADC2 Voltage 0V

When i connect the battery to GPIO1, while the USB is connected.
ADC1 Voltage 2V
ADC2 Voltage 3V

> There were pretty recent changes around how ADC is handled, better use latest development binary.

I did not use a precompiled binary, i compiled it with gitpod.io,
like suggested here:
https://tasmota.github.io/docs/Gitpod/
> 
> You also have the relatively new command `AdcGpio`, which you can use to customize details, after already having a value, with same linear fitting as the older `AdcParam` command.

Here is the output with USB and battery on GPIO1:
19:19:23.976 CMD: AdcGpio
19:19:23.981 MQT: stat/tasmota_07AEA8/RESULT = {"AdcGpio1":[0,4095,0,3.3]}

Here without usb and battery on GPIO0
19:29:52.709 CMD: AdcGpio0
19:29:52.714 MQT: stat/tasmota_07AEA8/RESULT = {"AdcGpio0":[0,4095,0,3.3]}
